python/ray/test/test_global_state.py

```diff
@@ -2,57 +2,57 @@
 from __future__ import division
 from __future__ import print_function
 
+import pytest
 import time
 
 import ray
 
 
-def setup_module():
-    if not ray.worker.global_worker.connected:
-        ray.init(num_cpus=1)
+@pytest.fixture
+def ray_start():
+    # Start the Ray processes.
+    ray.init(num_cpus=1)
+    yield None
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
 
-    # Finish initializing Ray. Otherwise available_resources() does not
-    # reflect resource use of submitted tasks
-    ray.get(cpu_task.remote(0))
 
+def test_replenish_resources(ray_start):
+    cluster_resources = ray.global_state.cluster_resources()
+    available_resources = ray.global_state.available_resources()
+    assert cluster_resources == available_resources
 
-@ray.remote(num_cpus=1)
-def cpu_task(seconds):
-    time.sleep(seconds)
+    @ray.remote
+    def cpu_task():
+        pass
 
+    ray.get(cpu_task.remote())
+    start = time.time()
+    resources_reset = False
 
-class TestAvailableResources(object):
     timeout = 10
-
-    def test_no_tasks(self):
-        cluster_resources = ray.global_state.cluster_resources()
+    while not resources_reset and time.time() - start < timeout:
```

Comment:
if this test is being modified, what are thoughts on using the pytest timeout? Seems more standard than manual implementation of timeout.

Reply:
I gave it a quick try but didn't get it working. I'd prefer to defer this to a different PR.